web/quiz_views.py

```
@@ -851,3 +1083,127 @@
+                            current_score += q.points
+                
+                # Update user quiz 
+                user_quiz.answers = json.dumps(answers)
+                user_quiz.score = (current_score / total_points * 100) if total_points > 0 else 0
+                
+                # Check if all questions are graded
+                all_graded = True
+                for q in quiz.questions.all():
+                    q_id = str(q.id)
+                    if q_id in answers:
+                        if not answers[q_id].get('is_graded', False) and not answers[q_id].get('is_correct', False):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)